kombu/transport/redis_cluster.py

```diff
@@ -147,11 +147,20 @@ def _register(self, channel, client, conn, cmd):
             self._unregister(*ident)
 
         if not conn.client:
-            if conn.key in channel.ask_errors:
-                ask_error = channel.ask_errors[conn.key]
-                node = channel.client.get_node(ask_error.host, ask_error.port)
-            else:
-                node = channel.client.get_node_from_key(conn.key)
+            tries = 0
+            while True:
+                if tries > 3:
+                    raise ValueError('Cannot find node for key: {}'.format(conn.key))
+                if conn.key in channel.ask_errors:
+                    ask_error = channel.ask_errors[conn.key]
+                    node = channel.client.get_node(ask_error.host, ask_error.port)
+                else:
+                    node = channel.client.get_node_from_key(conn.key)
```

Comment:
Yes, if a exception thrown from here, it'll be retried on next poll

Reply:
I think what @zach-iee wants to say is that we get `SlotNotCoveredError` here, we should reinitialize topology of redis cluster, and I can't see reinitialization for that case

Reply:
i don't understand the structure of the kombu completely, but in general, we should make sure that there is only a single place for handling errors that relates to the topology. If not, it will be hard to unit-test all cases using test codes.